{
    "cve_id": "CVE-2019-19274",
    "cve_description": "typed_ast 1.3.0 and 1.3.1 has a handle_keywordonly_args out-of-bounds read. An attacker with the ability to cause a Python interpreter to parse Python source (but not necessarily execute it) may be able to crash the interpreter process. This could be a concern, for example, in a web-based service that parses (but does not execute) Python code. (This issue also affected certain Python 3.8.0-alpha prereleases.)",
    "cve_publish_date": "2019-11-26",
    "cwe_id": "CWE-125",
    "cwe_name": "Out-of-bounds Read",
    "cwe_description": "The product reads data past the end, or before the beginning, of the intended buffer.",
    "commit_message": "bpo-35766: Merge typed_ast back into CPython (GH-11645)",
    "type_of_change": "Modification",
    "filename_of_changes": "Grammar",
    "code_language": "Python",
    "number_of_lines_added_for_mitigation": "22",
    "number_of_lines_deleted_vulnerable_to_cve": "9",
    "vulnerable_lines": [
        "// Line_Reference 20: funcdef: 'def' NAME parameters ['->' test] ':' suite",
        "// Line_Reference 23: typedargslist: (tfpdef ['=' test] (',' tfpdef ['=' test])* [',' [",
        "// Line_Reference 24:         '*' [tfpdef] (',' tfpdef ['=' test])* [',' ['**' tfpdef [',']]]",
        "// Line_Reference 25:       | '**' tfpdef [',']]]",
        "// Line_Reference 26:   | '*' [tfpdef] (',' tfpdef ['=' test])* [',' ['**' tfpdef [',']]]",
        "// Line_Reference 27:   | '**' tfpdef [','])",
        "// Line_Reference 42:                      ('=' (yield_expr|testlist_star_expr))*)",
        "// Line_Reference 74: for_stmt: 'for' exprlist 'in' testlist ':' suite ['else' ':' suite]",
        "// Line_Reference 80: with_stmt: 'with' with_item (',' with_item)*  ':' suite"
    ]
}
